src/plugins/form/directive.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  Directive,
+  Input,
+  OnInit,
+  OnDestroy,
+  ChangeDetectorRef
+} from "@angular/core";
+import { FormGroupDirective } from "@angular/forms";
+import { Store } from "../../store";
+import { Subject } from "rxjs/Subject";
+import { takeUntil } from "rxjs/operators/takeUntil";
+import { debounceTime } from "rxjs/operators/debounceTime";
+import {
+  UpdateFormStatus,
+  UpdateFormValue,
+  UpdateFormDirty,
+  UpdateFormErrors,
+  UpdateForm
+} from "./actions";
+
+const getValue = (obj, prop) =>
+  prop.split(".").reduce((acc, part) => acc && acc[part], obj);
+
+@Directive({ selector: "[ngxsForm]" })
+export class FormDirective implements OnInit, OnDestroy {
+  @Input("ngxsForm") path: string;
+  @Input("ngxsFormDebounce") debounce = 100;
+  @Input("ngxsFormClearOnDestroy") clearDestroy: boolean;
+
+  private _destroy$ = new Subject<null>();
+  //  private _updating = false;
+
+  constructor(
+    private _store: Store,
+    private _formGroupDirective: FormGroupDirective,
+    private _cd: ChangeDetectorRef
+  ) {}
+
+  ngOnInit() {
+    this._store
+      .select(state => getValue(state, `${this.path}.model`))
+      .pipe(takeUntil(this._destroy$))
+      .subscribe(model => {
+        //        if (!this._updating) {
```

Comment:
This updating flag is here to prevent circular loops. Why did you remove it?

Reply:
This was causing an issue when I update a form's state in an action and it was not reflecting back in the form. After I commented this code, the issue was resolved and I don't see any circular loops.